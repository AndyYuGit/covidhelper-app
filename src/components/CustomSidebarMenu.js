import React , { useEffect }from 'react';
import {
  SafeAreaView,
  View,
  StyleSheet,
  Image,
  Text,
  Linking,
} from 'react-native';
 
import {
  DrawerContentScrollView,
  DrawerItemList,
  DrawerItem,
} from '@react-navigation/drawer';
import { useDrawerStatus } from '@react-navigation/drawer';

import { useNavigation} from '@react-navigation/native';
import Constants from "expo-constants"
import {Logging} from '../common/Logging'

export function CustomSidebarMenu(props){
    const navigation = useNavigation();
    const version = Constants.manifest.version
    const isDrawerOpen = useDrawerStatus() === 'open';

    useEffect(() => {
        if(isDrawerOpen){
            Logging('Open the side bar menu','INFO')  
        }
    },[isDrawerOpen])

    return(
        <SafeAreaView style={{flex: 1}}>     
            <DrawerContentScrollView {...props} contentContainerStyle={{paddingTop: 100}}>
                <View style={styles.scrollHeader}>
                    <Text style={{
                    fontSize: 16,
                    textAlign: 'center',
                    color: 'white',
                    margin:40,
                    fontWeight: 'bold',
                    }}>
                        Covid Helper v{version}
                    </Text>
                </View>
                <DrawerItem
                labelStyle={styles.drawerItemLabelStyle}
                label="Report"
                icon={()=>{
                    return(
                        <Image source={require('../../assets/Text-Document-icon.png')}/>
                    );
                }}
                onPress={async ()=>{
                    navigation.navigate('Report', { screen: 'ReportScreen' });
                }}
                />
                <DrawerItem
                labelStyle={styles.drawerItemLabelStyle}
                label="Contact"
                icon={()=>{
                    return(
                        <Image source={require('../../assets/mailbox-icon.png')}/>
                    );
                }}
                onPress={async ()=>{
                    navigation.navigate('Contact', { screen: 'ContactScreen' });
                }}
                />
            </DrawerContentScrollView>
        </SafeAreaView>
    );
};
 

const styles = StyleSheet.create({
    container: {
      flex: 1,
      padding: 20,
      flexDirection:"row",
    },
    drawerItemLabelStyle:{
        color: '#21254a',
    },
    scrollHeader:{
        backgroundColor:'#21254a',
    }
  });

export default CustomSidebarMenu;