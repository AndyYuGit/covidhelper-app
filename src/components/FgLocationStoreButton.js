import { StyleSheet, Text, View ,Switch,TouchableOpacity, Alert,ActivityIndicator} from 'react-native';
import AsyncStorage from "@react-native-async-storage/async-storage";
import * as Location from 'expo-location';
import React, { useEffect, useState } from "react";
import { NavigationContainer, useIsFocused } from '@react-navigation/native';
import {Logging} from '../common/Logging'

export function FgLocationStoreButton(props){
    const isFocused = useIsFocused();
    const [showLoading, setShowLoading]=useState(false)

    async function storeLocation(){
        Logging('Store location start','INFO')
        setShowLoading(true)
        const {status:fgStatus} =await Location.requestForegroundPermissionsAsync();
        if (fgStatus != 'granted') {
            Logging('Permission to access location was denied','INFO')
            return;
        }else{
            try {
              let tempLocations=await AsyncStorage.getItem('storedLocations')
              if(tempLocations==null){
                tempLocations=[]
              }
              else{
                tempLocations=JSON.parse(tempLocations)
              }  
              let tempObject=new Object();
              let tempDate=new Date()
              let currentLocation=await Location.getCurrentPositionAsync({
                accuracy: Location.Accuracy.BestForNavigation,
              })
              tempObject.latitude=currentLocation.coords.latitude
              tempObject.longitude=currentLocation.coords.longitude
              tempObject.year=tempDate.getFullYear();
              tempObject.month=tempDate.getMonth()+1;
              tempObject.date=tempDate.getDate();
              tempObject.name=props.name
              tempLocations.push(tempObject)
              tempLocations=JSON.stringify(tempLocations)
              await AsyncStorage.setItem('storedLocations', tempLocations);
              setShowLoading(false)
              Logging('Location '+tempObject.latitude+' '+tempObject.longitude+' is stored','INFO')
            } catch (error) {
              Logging('Location store is failed:'+error,'ERROR')
            }
            
        }
    }

    useEffect(() => {
        
    },[isFocused]);

    return (
          <View>
            {showLoading||
            <TouchableOpacity
              style={styles.button}
              onPress={storeLocation}
            >
              <Text style={{color:'#ffffff',}}>Store</Text>
            </TouchableOpacity>}
            {showLoading&&<ActivityIndicator size="small" color="#cba3ce" />}
          </View>
          
      );
}

const styles = StyleSheet.create({
    button:{
      fontSize:14,
      backgroundColor:'#21254a',
      borderRadius:10,
      height:25,
      borderColor:'#cba3ce',
      borderWidth: 2.4,
    },
    
});