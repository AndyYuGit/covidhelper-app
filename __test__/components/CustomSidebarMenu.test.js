import React from 'react';
import renderer from 'react-test-renderer';
import {CustomSidebarMenu} from "../../src/components/CustomSidebarMenu";
import 'react-native-gesture-handler/jestSetup';

jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');
  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.useFakeTimers()

jest.mock('@react-navigation/drawer', () => ({
    useDrawerStatus: jest.fn(),
    addEventListener: jest.fn(),
    createDrawerNavigator: jest.fn(),
    DrawerContentScrollView:jest.fn(),
    DrawerItem:jest.fn(),
  }));

  jest.mock('@react-navigation/native', () => ({
    useNavigation: jest.fn(),
  }));

test('renders correctly', () => {
    const tree = renderer.create(<CustomSidebarMenu />).toJSON();
    expect(tree).toMatchSnapshot();
  });