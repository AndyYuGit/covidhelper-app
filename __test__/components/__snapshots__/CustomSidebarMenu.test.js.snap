// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
